testing/multi-env-test-engine/src/main/java/org/projectnessie/junit/engine/ThreadPerTestClassExecutionTestEngine.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.junit.engine;
+
+import java.util.Optional;
+import org.junit.jupiter.engine.config.CachingJupiterConfiguration;
+import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
+import org.junit.jupiter.engine.config.JupiterConfiguration;
+import org.junit.jupiter.engine.descriptor.JupiterEngineDescriptor;
+import org.junit.jupiter.engine.discovery.DiscoverySelectorResolver;
+import org.junit.jupiter.engine.execution.JupiterEngineExecutionContext;
+import org.junit.jupiter.engine.support.JupiterThrowableCollectorFactory;
+import org.junit.platform.engine.EngineDiscoveryRequest;
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine;
+import org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutorService;
+import org.junit.platform.engine.support.hierarchical.ThrowableCollector;
+
+public class ThreadPerTestClassExecutionTestEngine
+    extends HierarchicalTestEngine<JupiterEngineExecutionContext> {
+
+  @Override
+  public String getId() {
+    return JupiterEngineDescriptor.ENGINE_ID;
+  }
+
+  /** Returns {@code org.junit.jupiter} as the group ID. */
+  @Override
+  public Optional<String> getGroupId() {
+    return Optional.of("org.junit.jupiter");
+  }
+
+  /** Returns {@code junit-jupiter-engine} as the artifact ID. */
+  @Override
+  public Optional<String> getArtifactId() {
+    return Optional.of("junit-jupiter-engine");
```

Comment:
I'm not sure this is correct. Do we _have to_ use this ID to make stuff work?

Reply:
I mean: would a Nessie artifact ID work here?

Reply:
The ID doesn't really matter. And with the Junit-PR this would go away entirely.

Reply:
what's the Junit PR?

Reply:
https://github.com/projectnessie/nessie/pull/9453#discussion_r1739658367

Reply:
It's an issue ;)

Reply:
https://github.com/junit-team/junit5/pull/3941